### PR TITLE
Mock of pinpoint grpc with mockgen

### PR DESCRIPTION
**Pull request by suujia**
:tickets: **Ticket(s)**: #5 
---

## :construction_worker: Changes

In cases where the core gRPC server is down or busy, this acts as a mock, stand-in core client that performs the same functionality. This is great for testing client-side logic without connecting to a real server.`PinpointCoreClient` in this case is the client interface, and this mock service implements it. 

## :flashlight: Testing Instructions

🤷🏻‍♀️

### CODE REVIEW

**Review by bobheadxi — COMMENTED** [review]
This is good! Some things to wrap up this PR:

- [ ] a test to demonstrate use of this (there is currently an "integrated" test - one that calls the gRPC server - but it would be good to have a proper "unit" test as well, which is one that uses a mock instead!)
- [ ] a `make` script to update the mock as the gRPC interface evolves

**Comment by bobheadxi** [comment]
Also, travis reports the following error:

```
can't load package: core/main.go:10:2: found packages mock_pinpoint (mockpinpoint.pb.go) and pinpoint (pinpoint.pb.go) in /home/travis/gopath/src/github.com/ubclaunchpad/pinpoint/grpc
```

You cannot have multiple package names in a directory - if you use visual studio code, it probably should have warned you about this

File: grpc/mockpinpoint.pb.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
+// Code generated by MockGen. DO NOT EDIT.
```
Comment (leesw98) [inline grpc/mockpinpoint.pb.go]: How is this generated?

Comment (bobheadxi) [reply grpc/mockpinpoint.pb.go]: [mockgen](https://github.com/golang/mock) - but yeah, this PR needs reproducibility, documentation, etc.

**Pull request closed without merging**